homeassistant/data_entry_flow.py

```diff
@@ -301,19 +319,33 @@ def async_abort(self, flow_id: str) -> None:
     @callback
     def _async_add_flow_progress(self, flow: FlowHandler) -> None:
         """Add a flow to in progress."""
+        if flow.init_data is not None:
+            init_data_type = type(flow.init_data)
```

Comment:
Why use the type and not the source?

Reply:
HomeKit and Zeroconf use the same discovery type.

Reply:
I can move it one level up into `ConfigEntriesFlowManager`

Reply:
It doesn't require a source but in that case source is None: https://github.com/home-assistant/core/blob/dev/homeassistant/data_entry_flow.py#L399-L402 

Type works I guess. It's just another way of doing things.